packages/odyssey-react-mui/src/labs/DataComponents/TableContent.tsx

```diff
@@ -0,0 +1,384 @@
+    pageSize: number;
+  };
+  rowReorderingUtilities: {
+    dragHandleStyles: CSSObject;
+    dragHandleText: {
+      title: string;
+      "aria-label": string;
+    };
+    draggableTableBodyRowClassName: ({
+      currentRowId,
+      draggingRowId,
+      hoveredRowId,
+    }: {
+      currentRowId: string;
+      draggingRowId?: string;
+      hoveredRowId?: string;
+    }) => string | undefined;
+    handleDragHandleKeyDown: ({
+      table,
+      row,
+      event,
+    }: {
+      table: MRT_TableInstance<MRT_RowData>;
+      row: MRT_Row<MRT_RowData>;
+      event: React.KeyboardEvent<HTMLButtonElement>;
+    }) => void;
+    handleDragHandleOnDragCapture: (
+      table: MRT_TableInstance<MRT_RowData>,
+    ) => void;
+    handleDragHandleOnDragEnd: (table: MRT_TableInstance<MRT_RowData>) => void;
+    resetDraggingAndHoveredRow: (table: MRT_TableInstance<MRT_RowData>) => void;
+    updateRowOrder: ({
+      rowId,
+      newRowIndex,
+    }: {
+      rowId: string;
+      newRowIndex: number;
+    }) => void;
+  };
+  rowSelection: MRT_RowSelectionState;
+  setRowSelection: Dispatch<SetStateAction<MRT_RowSelectionState>>;
+  setTableState: Dispatch<SetStateAction<TableState>>;
+  tableOptions: TableProps;
+  tableState: TableState;
+  totalRows: UniversalProps["totalRows"];
+};
+
+const TableContent = ({
+  columns,
+  data,
+  draggingRow,
+  emptyState,
+  getRowId,
+  hasRowReordering,
+  hasRowSelection,
+  isEmpty,
+  isLoading,
+  isNoResults,
+  isRowReorderingDisabled,
+  onReorderRows,
+  pagination,
+  rowReorderingUtilities,
+  rowSelection,
+  setRowSelection,
+  setTableState,
+  tableOptions,
+  tableState,
+  totalRows,
+}: TableContentProps) => {
+  const [isTableContainerScrolledToStart, setIsTableContainerScrolledToStart] =
+    useState(true);
+  const [isTableContainerScrolledToEnd, setIsTableContainerScrolledToEnd] =
+    useState(true);
+  const [tableInnerContainerWidth, setTableInnerContainerWidth] =
+    useState<string>("100%");
+  const tableOuterContainerRef = useRef<HTMLDivElement>(null);
+  const tableInnerContainerRef = useRef<HTMLDivElement>(null);
+  const tableContentRef = useRef<HTMLTableElement>(null);
+
+  useScrollIndication({
+    setIsTableContainerScrolledToEnd: setIsTableContainerScrolledToEnd,
+    setIsTableContainerScrolledToStart: setIsTableContainerScrolledToStart,
+    setTableInnerContainerWidth: setTableInnerContainerWidth,
+    tableInnerContainer: tableInnerContainerRef.current,
+    tableOuterContainer: tableOuterContainerRef.current,
+  });
+
+  const odysseyDesignTokens = useOdysseyDesignTokens();
+  const { t } = useTranslation();
+
+  const columnIds = useMemo(() => {
+    return columns.map((column) => column.accessorKey) ?? [];
+  }, [columns]);
+
+  const columnOrder = useMemo(
+    () => [
+      "mrt-row-drag",
+      "mrt-row-select",
+      "mrt-row-expand",
+      ...columnIds,
+      "mrt-row-actions",
+    ],
+    [columnIds],
+  ) as string[];
```

Comment:
Can we avoid `string[]` here? It should be pretty clear this is a string array. If it's because `columnIds` can have `null` or `undefined`, we can filter them out.